api/answer.go

```diff
@@ -32,15 +30,120 @@ var (
   select   *
   from     answers
   full join     (%s) on answer = answers.id
-  where    deleted_at is NULL and answers.parent is NULL and answers.question = ? 
+  where    deleted_at is NULL and answers.parent is NULL and answers.question = ?
 `, VOTES_QUERY)
 	REPLIES_QUERY = `
   select   *
   from     answers
   where    deleted_at is NULL and answers.parent IN ?
 `
+	names = []string{"Wyatt", "Vivian", "Maria", "Alexander", "Luis", "Aidan", "Mason", "Aiden", "Mackenzie", "Adrian", "Oliver", "Andrea", "Amaya", "Nolan", "Riley", "Robert", "Ryker", "Sara", "Ryan", "Sawyer"}
 )
 
+func ConvertAnswerToAPI(answer Answer, id uint) (*AnswerResponse, error) {
+	db := util.GetDb()
+	usr, err := GetUserByID(db, id)
+	if err != nil {
+		return nil, err
+	}
+
+	var avatar, username string
+
+	if answer.Anonymous {
+		avatar = util.GenerateAnonymousAvatar(usr.Alias)
+		username = usr.Alias
+	} else {
+		avatar = fmt.Sprintf("https://avatars.githubusercontent.com/u/%d?v=4", usr.ID)
+		username = usr.Username
+	}
+
+	// recursively convert replies
+	var replies []AnswerResponse
+	for _, reply := range answer.Replies {
+		reply, err := ConvertAnswerToAPI(reply, id)
+		if err != nil {
+			return nil, err
+		}
+		replies = append(replies, *reply)
+	}
+
+	return &AnswerResponse{
+		ID:        answer.ID,
+		CreatedAt: answer.CreatedAt,
+		UpdatedAt: answer.UpdatedAt,
+		Question:  answer.Question,
+		Parent:    answer.Parent,
+		User:      username,
+		Content:   answer.Content,
+		Upvotes:   answer.Upvotes,
+		Downvotes: answer.Downvotes,
+		Replies:   replies,
+		Anonymous: answer.Anonymous,
+		AvatarURL: avatar,
+	}, nil
+
+}
+
+func GetUserByID(db *gorm.DB, id uint) (*User, error) {
+	var user User
+	if err := db.First(&user, id).Error; err != nil {
+		return nil, err
+	}
+	return &user, nil
+}
+
+func GetOrCreateUserByID(db *gorm.DB, id uint, username string) (*User, error) {
+	user, err := GetUserByID(db, id)
+	if err == nil {
+		return user, nil
+	}
+	if err != gorm.ErrRecordNotFound {
```

Comment:
È una cosa veramente minuscola, ma metterei un `else if`:
```suggestion
	} else	if err != gorm.ErrRecordNotFound {
```